src/Angor/Client/Pages/Browse.razor

```diff
@@ -215,11 +216,93 @@ else
     protected override async Task OnInitializedAsync()
     {
         projects = SessionStorage.GetProjectIndexerData() ?? new List<ProjectIndexerData>();
-        if (!projects.Any())
+
+        if (projects.Any())
+        {
+            // Projects exist, initialize the search focus
+            await JS.InvokeVoidAsync("angor.initSearchFocus", "searchQuery");
+
+            // Check for indexer and relay availability in the background
+            _ = CheckIndexerAndRelayAvailability();
+        }
+        else
+        {
+            // No projects are cached, fetch new projects after ensuring services are available
+            await FetchAndEnsureServices();
+        }
+    }
+
+    private async Task FetchAndEnsureServices()
+    {
+        try
         {
+            _NetworkService.AddSettingsIfNotExist();
+            await _NetworkService.CheckServices(true);
```

Comment:
here this is the right way I believe it will call the http api if last time it was called was more then x seconds